quarkus/runtime/src/main/java/org/keycloak/configuration/PropertyMappers.java

```diff
@@ -133,6 +131,19 @@ private static void configureDatabasePropertyMappers() {
         createWithDefault("db.pool.max-size", "quarkus.datasource.jdbc.max-size", String.valueOf(100), "The maximum size of the connection pool.");
     }
 
+    private static void configureClustering() {
+        createWithDefault("cluster.enabled", "kc.spi.connections-infinispan.default.clustered", "placeholder", (value, context) -> {
```

Comment:
@pedroigor I needed to use "placeholder" for the defaultValue as the transformer function is not executed in case that value is null. If we want to remove the workaround with "placeholder", we may need to change this line and remove the need to have non-null defaultValue - https://github.com/keycloak/keycloak/blob/master/quarkus/runtime/src/main/java/org/keycloak/configuration/PropertyMapper.java#L137 .

Then it will be possible to run transformer function even if "value" is null - which I am not 100% sure we want to support? WDYT?

Reply:
I think we can pass null to the transformer and delegate the default value resolution to it.